CONTRIBUTING.md

```diff
@@ -0,0 +1,30 @@
+# How to successfully make changes to Ninja
+
+We're very wary of changes that increase the complexity of Ninja (in particular,
+new build file syntax or command-line flags) or increase the maintenance burden
+of Ninja. Ninja is already successfully used by hundreds of developers for large
+projects and it already achieves (most of) the goals we set out for it to do.
+It's probably best to discuss new feature ideas on the
+[mailing list](https://groups.google.com/forum/#!forum/ninja-build) or in an
+issue before creating a PR.
+
+## Coding guidelines
+
+Generally it's the
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) with
+a few additions:
+
+* Ninja should also compile as C++03. You may use C++11 features in a test or an
+  unimportant tool if you guard your code with `#if __cplusplus > 201103L`.
```

Comment:
Note:

This does not explain

1) WHY
2) What the criteria is for Ninja to switch to something that is not nearly 20 years old.

Please indicate the motivations behind this requirement, and the timeline for the Ninja codebase modernizing.

So many PRs that are submitted to Ninja include C++11 only changes.

I continue to advocate that it's time to change to a modern version of the language, but at the *very least* you owe it to contributors to tell them the justification for using the old C++ standard, and tell them a timeline for when the project will modernize.

Reply:
I feel your pain. But the current HACKING.md doesn't explain any of this either. So the status quo isn't changed and I would rather see this small improvement right now, instead of chasing the perfect solution.

Reply:
I have updated the PR with your wording.

Reply:
I'm all for not chasing a perfect solution, but I don't think it's reasonable to not provide a couple of sentences explaining the rational behind using an older C++ release.

I'm not going to quibble on whatever wording you want to use, as much as I disagree with the motivation and decision. You're the project maintainer, so word things how you want. But it's not reasonable to expect contributors to read all issues that the project has ever had here on github (open, or closed) to understand the history behind the choice.

You just participated in a PR where someone explicitly used C++11 functionality earlier this week, and there were (if memory serves) two others in the last 30 days, which I think demonstrates the need for this kind of project policy documentation.

At the very least people reviewing code can provide a link to the CONTRIBUTING.md file to justify why C++11 only code is being rejected.

Example wording:

Rational:

> The Ninja project uses C++03, and not a newer C++ standard, to maintain support for ancient operating systems that will never have access to a C++ compiler that supports a C++ standard newer than C++03.

Timeline:

> The Ninja project currently has no plans to change from C++03 to a newer C++ standard.